t/03-testapi.t

```diff
@@ -580,6 +580,14 @@ subtest 'validate_script_output' => sub {
         },
         qr/coderef or regexp/
     );
+
+    my $exp_args = [123, proceed_on_failure => 1, type_command => 1];
+    $mock_testapi->redefine(script_output => sub {
+            my ($script, @args) = @_;
+            is_deeply(\@args, $exp_args, 'Arguments successful passed');
+            return 'foo';
+    });
+    validate_script_output('script', qr/foo/, @$exp_args);
```

Comment:
We should use proper function signatures where we can. We have converted all tests to use signatures but not yet the testapi.pm module itself. Please try if this works

```suggestion
    $mock_testapi->redefine(script_output => sub ($script, @args) {
            is_deeply(\@args, $exp_args, 'Arguments successful passed');
            return 'foo';
    });
    validate_script_output('script', qr/foo/, @$exp_args);
```

but I recommend against putting Test::More functions into mocked functions because on failure the output would point to the source of the mocked function, not to where this is called. So better to avoid actual test instructions within any mocked or actually any helper functions and just use them only on the highest level of the test.

Reply:
done